File: solutions/Golang/single_number_xor.go
package main

import "fmt"

func singleNumberXor(nums []int) int {
	result := 0
	for _, num := range nums {
		result ^= num
	}
	return result
}

func main() {
	fmt.Println(singleNumberXor([]int{2, 2, 1}))  // 1
	fmt.Println(singleNumberXor([]int{4, 1, 2, 1, 2}))  // 4
}

File: solutions/Java/single_number_xor.java
public class SingleNumberXor {
    public static int singleNumberXor(int[] nums) {
        int result = 0;
        for (int num : nums) {
            result ^= num;
        }
        return result;
    }

    public static void main(String[] args) {
        int[] nums = {2, 2, 1};
        System.out.println(singleNumberXor(nums));  // 1
    }
}
